### ISSUE TITLE
Controllable and Interpretable Singing Voice Decomposition via Assem-VC

### ISSUE THREAD

**Issue by AK391** [mindslab-ai/assem-vc#27]
just saw this paper https://arxiv.org/abs/2110.12676, when will the repo be updated for this thanks

**Comment by iehppp2010** [comment]
I have tried to reproduce this paper.
My train mse mel loss reaches 0.18, while dev mel loss stops at 0.75. 
![image](https://user-images.githubusercontent.com/22955379/147548036-9d7e4fe8-bf0b-4c0b-b23e-3d4694421ba1.png)

After train the 'Decoder' model, I use this model to do GTA fine finetuning on the  HiFi-GAN model you provide. 
Below is the HiFi-GAN model fine tuning loss
![image](https://user-images.githubusercontent.com/22955379/147548511-e1b45aa7-af28-45c1-bf86-5f72902c5f7e.png)

After that, I try to control speaker Identity by just switching the speaker embedding to target speaker, which is the the 
way you said in the paper.
I use a trained audio of  CSD female speaker as the reference audio(the link below).
https://drive.google.com/file/d/1QCGlfREai1AgkKnrLhdvZm-jt_k50R79/view?usp=sharing
I use the speaker PAMR in NUS-48E dataset as target speaker.
https://drive.google.com/file/d/19eL1XgAjR4eWTFv7M5jaMJMCWIC17m36/view?usp=sharing
The result audio is:
https://drive.google.com/file/d/1XsaWrSQ2xtiohbjpm6fFU-V28o4pp2wM/view?usp=sharing

**I found that lyrics are hard to hear clearly.**

My dataset config:
devset: 
   CSD speaker, these three audio  en48/en49/en50 were chosed;
   NUS-48E speakers, ADIZ's 13 and JLEE 05 were chosed.
trainset: the other songs in CSD and NUS-48E.

My speaker embedding dimension is 256.( It seems 256 is too large?)

**I want to know what could  be the problem with my model?**
And can you share you Decoder model train/dev loss? 
My Decoder model got a relative larger mel MSE loss on devset than trainset.

**Comment by wookladin** [comment]
@iehppp2010 
Hi. I think your alignment encoder, 'Cotatron' doesn't seem to be working properly.
As explained in the paper, we transferred Cotatron from pre-trained weights, which are trained with LibriTTS and VCTK.
Did you transfer from those weights?
You can find pre-trained weights in this [Google Drive link](https://drive.google.com/drive/folders/1aIl8ObHxsmsFLXBz-y05jMBN4LrpQejm).

**Comment by iehppp2010** [comment]
@wookladin 
Thanks for your quick reply.
I do used the your pre-trained weights.
When I train the 'Decoder' model, the 'Cotatron' aligner model is **freezed**.
I found the plotted alignment is not as good as other TTS model，e.g. Tacotron2.
![image](https://user-images.githubusercontent.com/22955379/147632196-af185f4a-d1fa-4e00-ae23-3a9b068e3995.png)

I want to know if I need to do fine-tune the 'Cotatron' model on singing dataset to get better alignment result?
Wish your reply.

**Comment by wookladin** [comment]
@iehppp2010 
Yes. 
You first have to fine-tune Cotatron model on the singing dataset, because the average duration of each phoneme is much longer in the singing dataset.
It would generate better alignment and sample quality

**Comment by iehppp2010** [comment]
> @iehppp2010 Yes. You first have to fine-tune Cotatron model on the singing dataset, because the average duration of each phoneme is much longer in the singing dataset. It would generate better alignment and sample quality

@wookladin 
Thanks for you quickly reply.
After I fine-tune the Cotatron model, the train.loss_reconstruction converges about 0.2, while val.loss_reconstruction
 got mininum value about 0.5 at step 3893.

![image](https://user-images.githubusercontent.com/22955379/148345957-12bce7ae-8cee-48c6-adf5-b632f27feeac.png)

I use that checkpiont to train the Decoder model and fine tune HIFI-GAN vocoder.

I found that when test with an audio if the fine-tuned Cotatron model never seen it, I can't get good sample quality. 
I guess it's the reason that Cotatron model gives not good alignment...

So, I want to  know  how to let the Cotatron model get better alignment on unseen sing audio?
Besides, can you provide more training details?

**Comment by betty97** [comment]
@iehppp2010, I am also trying to reproduce the results of this paper. I have one doubt regarding the dataset preparation: how did you split the files? In the paper it is said that "all singing voices are split between 1-12 seconds", did you do it manually for both CSD and NUS-48E, or how? Thanks!!
